shared/macros-ansible.jinja

```diff
@@ -346,3 +346,189 @@ The macro requires following parameters:
     create: yes
   when: find_existing_watch_audit_rules.matched is defined and find_existing_watch_audit_rules.matched == 0
 {{%- endmacro %}}
+
+{{#
+The following macro remediates Audit syscall rule in /etc/audit/rules.d directory.
+The macro requires following parameters:
+- syscalls: list of syscalls supplied as a list ["syscall1", "syscall2"] etc.
+- key: a key to use as rule identifier.
+Note that if there  already exists a rule wit the same key in the /etc/audit/rules.d directory, the rule will be placed in the same file.
+The rule determines the architecture of the system and apply appropriate remediations.
+It utilizes b32 for X86 architecture and both b32 and b64 for x86_64 architecture.
+#}}
+
+{{% macro ansible_audit_augenrules_add_syscall_rule(syscalls=[], key="") -%}}
+#
+# What architecture are we on?
+#
+- name: Set architecture for audit modules tasks
+  set_fact:
+    audit_arch: "b{{ ansible_architecture | regex_replace('.*(\\d\\d$)','\\1') }}"
+
+- name: Declare list of syscals
+  set_fact:
+    syscalls: {{{ syscalls }}}
+
+- name: Declare number of syscalls
+  set_fact: audit_syscalls_number_of_syscalls="{{ syscalls|length|int }}"
+
+- name: Check existence of syscalls for 32 bit architecture in /etc/audit/rules.d/
+  find:
+    paths: "/etc/audit/rules.d"
+    contains: '^[\s]*-a[\s]+always,exit[\s]+(?:.*-F[\s]+arch=b32[\s]+)(?:.*(-S[\s]+{{ item }}[\s]+|([\s]+|[,]){{ item }}([\s]+|[,]))).*(-k[\s]+|-F[\s]+key=)[\S]+[\s]*$'
+    patterns: "*.rules"
+  register: audit_syscalls_found_32_rules_d
+  loop: "{{ syscalls }}"
+
+- name: Get number of matched 32 bit syscalls in /etc/audit/rules.d/
+  set_fact: audit_syscalls_matched_32_rules_d="{{audit_syscalls_found_32_rules_d.results|sum(attribute='matched')|int }}"
+
+- name: Check existence of syscalls for 64 bit architecture in /etc/audit/rules.d/
+  find:
+    paths: "/etc/audit/rules.d"
+    contains: '^[\s]*-a[\s]+always,exit[\s]+(?:.*-F[\s]+arch=b64[\s]+)(?:.*(-S[\s]+{{ item }}[\s]+|([\s]+|[,]){{ item }}([\s]+|[,]))).*(-k[\s]+|-F[\s]+key=)[\S]+[\s]*$'
+    patterns: "*.rules"
+  register: audit_syscalls_found_64_rules_d
+  loop: "{{ syscalls }}"
+
+- name: Get number of matched 64 bit syscalls in /etc/audit/rules.d/
+  set_fact: audit_syscalls_matched_64_rules_d="{{audit_syscalls_found_64_rules_d.results|sum(attribute='matched')|int }}"
+
+- name: Search /etc/audit/rules.d for other rules with the key {{{ key }}}
+  find:
+    paths: "/etc/audit/rules.d"
+    recurse: no
+    contains: '(-F key=)|(-k\s+){{{ key }}}$'
```

Comment:
We need test scenarios for the augenrules case.
This regex doesn't seem to be working, the remediation is not adding the rule together with other rules with same key.

Reply:
Adding anchor to beginning of file and another parenthesis group solved for me:  '^.*((-F key=)|(-k\s+)){{{ key }}}$'

Reply:
Nice catch, thanks. I created test cases for augen-rules. However, I don't think we can test this as I believe Oval doesn't care into which file in rules.d directory the rule is placed.

Reply:
> However, I don't think we can test this as I believe Oval doesn't care into which file in rules.d directory the rule is placed.

Right, nice catch too, :).
As the remediation is not adding duplicates I don't consider this to be a blocking issue.